spec/features/date_selection_spec.rb

```diff
@@ -0,0 +1,195 @@
+  end
+
+  it 'renders data for the previous month when `last month` is selected' do
+    to = Time.zone.today.last_month.end_of_month
+    from = Time.zone.today.last_month.beginning_of_month
+    stub_for(from, to)
+
+    visit '/metrics/base/path?metrics[]=unique_pageviews&metrics[]=pageviews'
+    find('#date_range_last-month').click
+    click_button 'OK'
+    sleep 2
+    click_on 'Unique pageviews table'
+
+    expect_metrics_for_each_date_to_be_correct(from, to)
+  end
+
+  it 'renders data for the last 3 months when `last 3 months` is selected' do
+    to = Time.zone.today
+    from = Time.zone.today - 3.months
+    stub_for(from, to)
+
+    visit '/metrics/base/path?metrics[]=unique_pageviews&metrics[]=pageviews'
+    find('#date_range_last-3-months').click
+    click_button 'OK'
+    click_on 'Unique pageviews table'
+
+    expect_metrics_for_each_date_to_be_correct(from, to)
+  end
+
+  it 'renders data for the last 6 months when `last6 months` is selected' do
+    to = Time.zone.today
+    from = Time.zone.today - 6.months
+    stub_for(from, to)
+
+    visit '/metrics/base/path?metrics[]=unique_pageviews&metrics[]=pageviews'
+    find('#date_range_last-6-months').click
+    click_button 'OK'
+    click_on 'Unique pageviews table'
+
+    expect_metrics_for_each_date_to_be_correct(from, to)
+  end
+
+  it 'renders data for the last year when `last year` is selected' do
+    to = Time.zone.today
+    from = Time.zone.today - 1.year
+    stub_for(from, to)
+
+    visit '/metrics/base/path?metrics[]=unique_pageviews&metrics[]=pageviews'
+    find('#date_range_last-1-year').click
+    click_button 'OK'
+    sleep 2
+    click_on 'Unique pageviews table'
+
+    expect_metrics_for_each_date_to_be_correct(from, to)
+  end
+
+  it 'renders data for the last 2 years when `last 2 years` is selected' do
+    to = Time.zone.today
+    from = Time.zone.today - 2.years
+    stub_for(from, to)
+
+    visit '/metrics/base/path?metrics[]=unique_pageviews&metrics[]=pageviews'
+    find('#date_range_last-2-years').click
+    click_button 'OK'
+    click_on 'Unique pageviews table'
+
+    expect_metrics_for_each_date_to_be_correct(from, to)
+  end
+
+  def initial_page_stub
+    from = Time.zone.today - 30.days
+    to = Time.zone.today
+
+    content_data_api_has_metric(base_path: 'base/path',
+      from: from,
+      to: to,
+      metrics: %w[unique_pageviews pageviews number_of_internal_searches],
+      payload: {
+        base_path: '/base/path',
+        unique_pageviews: 145_000,
+        pageviews: 200_000,
+        title: "Content Title",
+        first_published_at: '2018-02-01T00:00:00.000Z',
+        public_updated_at: '2018-04-25T00:00:00.000Z',
+        primary_organisation_title: 'The ministry',
+        document_type: "news_story"
+      })
+
+    content_data_api_has_timeseries(base_path: 'base/path',
+      from: from,
+      to: to,
+      metrics: %w[unique_pageviews pageviews number_of_internal_searches],
```

Comment:
Is it ok to have three metrics, and return only two sets of series?

Reply:
It is OK as far as the tests are concerned as we stub the response but I have changed the payload to more accurately reflect what real life response